src/planning/test/tests.cpp

```diff
@@ -1,14 +1,30 @@
 #include <chrono>
 #include <fstream>
+#include <set>
 
 #include "gtest/gtest.h"
 #include "planning/global_path_planner.hpp"
 #include "planning/local_path_planner.hpp"
+#include "planning/path_smoothing.hpp"
+#include "planning/track.hpp"
 #include "rclcpp/rclcpp.hpp"
 #include "utils/files.hpp"
 
 using testing::Eq;
 
+bool customComparator(const std::pair<double,  double>& a,  const std::pair<double,  double>& b) {
```

Comment:
If it needs to be like this so be it, but it really seems like you could use a data type in this case i.e. instead of a vector of Pairs, use a vector of a class (like Position) that overloads the < operator. In any way, it is just my opinion, not necessarily better, just though it felt right.

Reply:
It definitely did not need to be like that but the formatting of the result of the Delaunay Triangulations into a vector of pairs was already implemented and it makes testing substantially easier.

Reply:
Agree on this. Why would it be easier?

Reply:
Na, thats valid